app/services/activity/tab.rb

```diff
@@ -0,0 +1,80 @@
+class Activity
+  class Tab
+    include Pundit
```

Comment:
I'm a bit uneasy with such a complex and clever abstraction of the activity tabs:

- use of `send`
- use of `instance_variable_get`
- dynamically generated routes

BUT: you explain that this is a reaction to real pain experienced maintaining these views, so yes, let's give it a go!